### PR TITLE
Bootstrap force upgrade

### PR DESCRIPTION
**Pull request by whoarethebritons**
https://ocd.appscale.com:8080/job/Daily%20Build/3319/

### CODE REVIEW

File: bootstrap.sh [hunk 1]
```diff
@@ -270,9 +270,29 @@ if [ -d /etc/appscale/certs ]; then
...
-        else
+        elif [ FORCE_UPGRADE = "N" ]; then
             (cd appscale; git pull)
             (cd appscale-tools; git pull)
+        else
+            RANDOM_KEY="$(echo $(date), $$|md5sum|head -c 6)-$(date +%s)"
+            if ! (cd appscale;
+                    git remote add -f appscale-bootstrap-"${RANDOM_KEY}" "${APPSCALE_REPO}";
+                    git checkout appscale-bootstrap-"${RANDOM_KEY}"/"${APPSCALE_BRANCH}"); then
+                echo "Please stash your local unsaved changes and checkout"\
```
Comment (whoarethebritons) [inline bootstrap.sh]: I would like to change this but I am not sure what to change it to.

"Please make sure the repository url is correct, the branch exists, and that you have stashed your local changes."
"e.g.: git stash, git remote add -t {remote_branch} {repo_name} {repository_url}; git checkout {remote_branch}"

File: bootstrap.sh [hunk 2]
```diff
@@ -270,9 +270,29 @@ if [ -d /etc/appscale/certs ]; then
...
                 exit 1
             fi
-        else
+        elif [ FORCE_UPGRADE = "N" ]; then
             (cd appscale; git pull)
             (cd appscale-tools; git pull)
+        else
+            RANDOM_KEY="$(echo $(date), $$|md5sum|head -c 6)-$(date +%s)"
+            if ! (cd appscale;
+                    git remote add -f appscale-bootstrap-"${RANDOM_KEY}" "${APPSCALE_REPO}";
```
Comment (whoarethebritons) [inline bootstrap.sh]: going to change this to
 `git remote add -t "${APPSCALE_BRANCH}" appscale-bootstrap-"${RANDOM_KEY}" "${APPSCALE_REPO}"`

File: bootstrap.sh (hunk 1 above)
Comment (cdonati) [inline bootstrap.sh]: Sure. That sounds reasonable.

Comment (obino) [inline bootstrap.sh]: that would do. Or 'please make sure you are on the unmodified branch (git will refuse to pull)'

**Comment by whoarethebritons** [comment]
@cdonati this is mainly for us and developers, it is the first step for testing the `appscale upgrade` without a git tag.

**Comment by whoarethebritons** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/3337/

**Pull request merged**